### PR TITLE
Adding Docs for New CI

### PR DESCRIPTION
**Pull request by VIGNESHinZONE**
- [x]  Add documentation to DeepChem README on new install procedure with extras
- [ ] Add docs for contributors on marking tests according to backend

Issue - https://github.com/deepchem/deepchem/issues/2588

cc: @ncfrey @rbharath

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,6 +78,37 @@ We recommend installing RDKit with deepchem if you use conda.
 conda install -y -c conda-forge rdkit
 ```
 
+### Latest version
```
Comment (rbharath) [inline README.md:81]: Let's just name this section "Nightly build version" and remove the old directions for nightly build version below

File: README.md [hunk 2]
```diff
@@ -76,6 +78,37 @@ We recommend installing RDKit with deepchem if you use conda.
 conda install -y -c conda-forge rdkit
 ```
 
+### Latest version
+
+For using general utilites like Molnet, Featurisers, Datasets, etc, then, you install deepchem via pip.
```
Comment (rbharath) [inline README.md:83]: Add the line "The nightly version is built by the HEAD of DeepChem" to start this section

File: README.md [hunk 3]
```diff
@@ -76,6 +78,37 @@ We recommend installing RDKit with deepchem if you use conda.
...
 
+### Latest version
+
+For using general utilites like Molnet, Featurisers, Datasets, etc, then, you install deepchem via pip.  
+
+```bash
+pip install --pre deepchem
+```
+Deepchem provides support for tensorflow, pytorch, jax and each require
+a induvidual pip Installation.
```
Comment (rbharath) [inline README.md:89]: Typo: individual

File: README.md [hunk 4]
```diff
@@ -76,6 +78,37 @@ We recommend installing RDKit with deepchem if you use conda.
...
+
+```bash
+pip install --pre deepchem[torch]
+```
+For using models with jax dependencies, you install using
+
+```bash
+pip install --pre deepchem[jax]
+```
+If GPU support is required, then make sure its cuda packages installed and then install the NN library using the below links before installing deepchem
```
Comment (rbharath) [inline README.md:106]: How about: "If GPU support is required, then make sure CUDA is installed and then install the desired deep learning framework using the links below before installing deepchem"

File: README.md [hunk 5]
```diff
@@ -76,6 +78,37 @@ We recommend installing RDKit with deepchem if you use conda.
...
+
+1. tensorflow - just cuda installed
+2. pytorch - https://pytorch.org/get-started/locally/#start-locally
+3. jax - https://github.com/google/jax#pip-installation-gpu-cuda
+
 ### Nightly build version
```
Comment (rbharath) [inline README.md:112]: This section can be removed

File: docs/source/get_started/installation.rst [hunk 6]
```diff
@@ -29,6 +29,42 @@ We recommend installing RDKit with deepchem if you use conda.
 
     conda install -y -c conda-forge rdkit
 
+Latest version
```
Comment (rbharath) [inline docs/source/get_started/installation.rst:32]: We should merge this with the nightly build docs as indicated in the other comments above

File: docs/source/get_started/installation.rst [hunk 7]
```diff
@@ -163,6 +199,55 @@ If you are using the Windows and the PowerShell:
     pytest -m "not slow" deepchem # optional
 
 
+From source lightweight guide
```
Comment (rbharath) [inline docs/source/get_started/installation.rst:202]: Should we merge these with the existing from source guide? It seems similar in complexity I think

**Review by rbharath — COMMENTED** [review]
Looking good, but have a few comments.

@arunppsg Could you review this PR as well to coordinate with your PR adding new contributor guidelines?

File: docs/source/get_started/installation.rst [hunk 8]
```diff
@@ -163,6 +199,55 @@ If you are using the Windows and the PowerShell:
...
+.. code-block:: bash
+
+    git clone https://github.com/deepchem/deepchem.git
+    cd deepchem
+
+We would advise all users to use conda environment, following below-
+
+.. code-block:: bash
+
+    conda create --name deepchem python=3.7
```
Comment (arunppsg) [inline docs/source/get_started/installation.rst:224]: We can recommend [python 3.8.](https://github.com/deepchem/deepchem/pull/2592#discussion_r667406513). We should also update this in other parts of README.md (see first line in this [section](https://github.com/deepchem/deepchem/#requirements))

File: docs/source/get_started/installation.rst (hunk 7 above)
Comment (VIGNESHinZONE) [reply docs/source/get_started/installation.rst:202]: Well that one uses install_deepchem_conda script and we will have to build a new conda script for an overall installation

File: docs/source/get_started/installation.rst (hunk 8 above)
Comment (VIGNESHinZONE) [reply docs/source/get_started/installation.rst:224]: Yeah sure, will do

**Comment by VIGNESHinZONE** [comment]
@arunppsg I still haven't added the details about contribution guidelines, where contributors would have to take care of the - 
1. Pytest markers depending on the environment they would be contributing to, similar to this line
https://github.com/deepchem/deepchem/blob/1916ff4e25bd1a45c35f44238e93f728e189912a/deepchem/models/tests/test_kerasmodel.py#L20-L21
2. Since, there are four separate environments it would also be the contributor's responsibility to include
    the dependencies for particular environments in deepchem/requirements [link](https://github.com/deepchem/deepchem/tree/master/requirements)

Since https://github.com/deepchem/deepchem/pull/2592 is already working on adding these contribution guidelines could you directly include these two points there itself.

**Comment by VIGNESHinZONE** [comment]
@rbharath I have included python 3.9 in the workflows and it seems to be running well.

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**